tests/x509/test_x509.py

```diff
@@ -1000,11 +1000,18 @@ def test_generalized_time_not_after_cert(self, backend):
         assert cert.version is x509.Version.v3
 
     def test_invalid_version_cert(self, backend):
-        cert = _load_cert(
-            os.path.join("x509", "custom", "invalid_version.pem"),
-            x509.load_pem_x509_certificate,
-            backend
-        )
+        try:
+            cert = _load_cert(
+                os.path.join("x509", "custom", "invalid_version.pem"),
+                x509.load_pem_x509_certificate,
+                backend
+            )
+        except ValueError:
```

Comment:
I don't suppose BoringSSL puts something on the error stack so that we can tell it's an `x509.InvalidVersion` and raise that?

Reply:
It does, but given that the rest of cryptography.io doesn't build with BoringSSL anyway, I figured it wasn't worth doing anything more complicated there. (Also still TBD whether the change will stick. If I have to revert it and try later, the error code will disappear.)

Right now the errors are just dropped, for better or worse. Dunno if that's expected:
https://github.com/pyca/cryptography/blob/master/src/cryptography/hazmat/backends/openssl/backend.py#L1194-L1199

Reply:
(If you all don't want to support this, that's totally fine. I can carry a patch on my end easily enough.)

Reply:
Ooof. We need to figure out a strategy either way, since this results in a coverage issue (we enforce 100% coverage).

@reaperhulk any obvious theories of how to handle this?